src/python/bot/tasks/testcase_manager.py

```diff
@@ -526,121 +548,144 @@ def test_for_crash_with_retries(testcase,
+        logs.log('Crash is reproducible.')
+        return True
 
-    state = crash_result.get_symbolized_data()
-    logs.log(
-        ('Crash occurred in {crash_time} seconds (round {round_number}). '
-         'State:\n{crash_state}').format(
-             crash_time=crash_time,
-             round_number=round_number,
-             crash_state=state.crash_state),
-        output=state.crash_stacktrace)
+    logs.log('Crash is not reproducible. Crash count: %d/%d.' % (crash_count,
+                                                                 round_number))
+    return False
 
-    # If we don't have an expected crash state, set it to the one from initial
-    # crash.
-    if not expected_state:
-      expected_state = state.crash_state
 
-    if crash_result.is_security_issue() != expected_security_flag:
-      logs.log('Detected a crash without the correct security flag.')
-      continue
+def test_for_crash_with_retries(testcase,
+                                testcase_path,
+                                test_timeout,
+                                http_flag=False,
+                                compare_crash=True):
+  """Test for a crash and return crash parameters like crash type, crash state,
+  crash stacktrace, etc."""
+  runner = TestcaseRunner(testcase.fuzzer_name, testcase_path, test_timeout,
+                          testcase.gestures, http_flag)
+  crash_retries = environment.get_value('CRASH_RETRIES')
+  if compare_crash:
+    expected_state = testcase.crash_state
+    expected_security_flag = testcase.security_flag
+  else:
+    expected_state = None
+    expected_security_flag = None
 
-    crash_comparer = CrashComparer(state.crash_state, expected_state)
-    if not crash_comparer.is_similar():
-      logs.log('Detected a crash with an unrelated state: '
-               'Expected(%s), Found(%s).' % (expected_state, state.crash_state))
-      continue
+  return runner.reproduce_with_retries(crash_retries, expected_state,
+                                       expected_security_flag,
+                                       testcase.flaky_stack)
 
-    crash_count += 1
-    if crash_count >= reproducible_crash_target_count:
-      logs.log('Crash is reproducible.')
-      return True
 
-  logs.log('Crash is not reproducible. Crash count: %d/%d.' % (crash_count,
-                                                               round_number))
-  return False
+def test_for_reproducibility(fuzzer_name, testcase_path, expected_state,
+                             expected_security_flag, test_timeout, http_flag,
+                             gestures):
+  """Test to see if a crash is fully reproducible or is a one-time crasher."""
+  runner = TestcaseRunner(fuzzer_name, testcase_path, test_timeout, gestures,
+                          http_flag)
+  crash_retries = environment.get_value('CRASH_RETRIES')
+  return runner.test_reproduce_reliability(crash_retries, expected_state,
```

Comment:
s/test_reproduce_reliability/reproduce_for_reliability 
to match style of reproduce_with_retries

Reply:
This is unchanged from before, and I'm not sure that reproduce_for_reliability is as clear. I think that "test_for_reproducibility" more clearly states the intention that we're seeing how reliable the reproducibility is?

Reply:
Leaving this comment open. Happy to rename everything if you still think reproduce_for_reliability is better!

Reply:
ok lets keep-as is.